src/main/java/net/tropicraft/core/common/dimension/feature/jigsaw/SpawnerProcessor.java

```diff
@@ -0,0 +1,81 @@
+package net.tropicraft.core.common.dimension.feature.jigsaw;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mojang.datafixers.Dynamic;
+import com.mojang.datafixers.types.DynamicOps;
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.EntityType;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.IWorldReader;
+import net.minecraft.world.gen.feature.template.IStructureProcessorType;
+import net.minecraft.world.gen.feature.template.PlacementSettings;
+import net.minecraft.world.gen.feature.template.StructureProcessor;
+import net.minecraft.world.gen.feature.template.Template;
+import net.tropicraft.core.common.entity.TropicraftEntities;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class SpawnerProcessor extends StructureProcessor {
+    // TODO fix
+    //public static final IStructureProcessorType TYPE = Registry.register(Registry.STRUCTURE_PROCESSOR, Constants.MODID + ":spawner_processor", SpawnerProcessor::new);
```

Comment:
What needs to be fixed?

Reply:
[20:04:07] [Server thread/ERROR] [minecraft/IDynamicDeserializer]: Unknown type tropicraft:spawner_processor, replacing with net.minecraft.world.gen.feature.template.NopProcessor@29226615

Reply:
Wouldn't that be fixed by registering it, uncommenting this line?

Reply:
I wish - I got this error when everything was hooked up. It needs a little more testing to see what went wrong, but I decided not to take it out for now since it's 99% done.